src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/AdapterUtil.cs

```diff
@@ -764,27 +764,44 @@ internal static Version GetAssemblyVersion()
 
         internal static bool IsAzureSynapseOnDemandEndpoint(string dataSource)
         {
-            return IsEndpoint(dataSource, ONDEMAND_PREFIX)
+            return IsEndpoint(dataSource, s_azureSqlServerOnDemandEndpoints)
                 || dataSource.Contains(AZURE_SYNAPSE)
```

Comment:
Could you also make a collection of Synapse endpoints while you're here?

Reply:
@cheenamalhotra Do you mean change this:

```c#
return IsEndpoint(dataSource, s_azureSqlServerOnDemandEndpoints)
    || dataSource.Contains(AZURE_SYNAPSE)
    || dataSource.Contains(FABRIC_DATAWAREHOUSE)
    || dataSource.Contains(PBI_DATAWAREHOUSE)
    || dataSource.Contains(PBI_DATAWAREHOUSE2)
    || dataSource.Contains(PBI_DATAWAREHOUSE3);

```
to 

`
return IsEndpoint(dataSource, s_azureSynapseOnDemandEndpoints);

`